pom.xml

```diff
@@ -36,68 +36,179 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <tez.version>0.9.1</tez.version>
-    <hive.version>2.3.7</hive.version>
+    <hive.version>2.3.9</hive.version>
     <license.maven.plugin.version>3.0</license.maven.plugin.version>
     <hive.execution.engine>mr</hive.execution.engine>
     <junit.jupiter.version>5.7.1</junit.jupiter.version>
     <mockito.version>3.8.0</mockito.version>
+    <log4j2.version>2.17.1</log4j2.version>
   </properties>
   
   <dependencies>
+    <dependency>
```

Comment:
Why the new explicit dependency on Guava?

Reply:
As I made these changes, I started getting ClassNotFound exception for `FluentIterable`.

On checking in waggle dance, I saw that we have had to pull in this dependency explicitly there as well.

https://github.com/ExpediaGroup/waggle-dance/blob/main/waggle-dance-core/pom.xml#L84

Reply:
OK, yes, HiveRunner does use Guava directly so this makes sense. I guess it was getting a transitive version via Hive before. We're going to have to be very careful with this one so that it doesn't clash and should aim to use the same version as Hive is using. I assume that's why you went with version `18.0`?

Reply:
Hive uses `14.0.1` but Cloverleaf uses 18.0 and WD uses 23.0.

Reply:
Hmmm, does everything work if you use `14.01`? If so I'd suggest using that version and putting a comment in the pom.xml describing which Hive dependency this lines up with. I think it's worth not using a different version in case there are classpath issues which Guava and Hive are notorious for. You can add it to the other PR you just opened.

Reply:
Yeah. it works with Guava 14.0.1 as well. I just upgraded it thinking if we are able to upgrade and it doesn't break anything downstream, then we should.

Reply:
Yeah, normally I'd agree with doing this but I've been bitten so many times by Hive version clashes that it's better to stick with what they have.